File: src/main/java/com/mobilesolutionworks/android/zero/CredentialManager.java
package com.mobilesolutionworks.android.zero;

import java.util.List;

/**
 * Created by yunarta on 5/5/17.
 */

public interface CredentialManager<C extends Credential> {

    void add(String name, C credential);

    void update(String name, C credential);

    C get(String name);

    List<C> list();
}

File: src/main/kotlin/com/mobilesolutionworks/android/struts/CredentialManager.kt
package com.mobilesolutionworks.android.struts

/**
 * Created by yunarta on 13/5/17.
 */

public interface CredentialManager<C : Credential> {

    fun add(name: String, credential: C)

    fun update(name: String, credential: C)

    operator fun get(name: String): C

    fun list(): List<C>
}